templates/engage/index.html

```diff
@@ -38,4 +64,37 @@ <h1>Ubuntu case studies, whitepapers and webinars</h1>
   </div>
 
 </section>
+<script>
+(function() {
+  const languageSelector = document.getElementById("language-selector");
+  const urlObj = new URL(window.location);
+
+  function handleFilter(key, el, url) {
+    if (!el) {
+      return;
+    }
+
+    if (url.searchParams.has(key)) {
+      el.value = url.searchParams.get(key);
+    }
+
+    el.addEventListener("change", (e) => {
```

Comment:
Why are you using JS to submit this form? Why not just have a submit button that posts to the same page?

Reply:
We are following the `/tutorials` page